content/guides/hot-module-replacement.md

```diff
@@ -18,125 +19,194 @@ related:
 
 W> __HMR__ is not intended for use in production, meaning it should only be used in development. See the [building for production guide](/guides/production) for more information.
 
-
 ## Enabling HMR
 
-This feature is great for productivity. Let's take a look at how to set it up with [webpack-dev-server](https://github.com/webpack/webpack-dev-server)...
-
-``` js
-const path = require('path');
-const webpack = require('webpack');
-
-module.exports = {
-  entry: './index.js',
+This feature is great for productivity. All we need to do is update our [webpack-dev-server](https://github.com/webpack/webpack-dev-server) configuration, and use webpack's built in HMR plugin.
 
-  plugins: [
-    new webpack.HotModuleReplacementPlugin() // Enable HMR
+``` diff
+  const path = require('path');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
++ const webpack = require('webpack');
+
+  module.exports = {
+    entry: {
+      app: './src/index.js',
+      print: './src/print.js'
+    },
+    devtool: 'inline-source-map',
+    devServer: {
+      contentBase: './dist',
++     hot: true
+    },
+    plugins: [
+      new HtmlWebpackPlugin({
+        title: 'Hot Module Replacement'
+      }),
++     new webpack.HotModuleReplacementPlugin()
   ],
-
   output: {
-    filename: 'main.js',
-    path: path.resolve(__dirname, 'dist'),
-    publicPath: '/'
-  },
-
-  devServer: {
-    hot: true, // Tell the dev-server we're using HMR
-    contentBase: path.resolve(__dirname, 'dist'),
-    publicPath: '/'
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
   }
 };
 ```
 
-Not too bad, huh? Let's test it out using `module.hot.accept`...
+You can also use the CLI to modify the [webpack-dev-server](https://github.com/webpack/webpack-dev-server) configuration with the following command: `webpack-dev-server --hotOnly`.
+
+Now let's update the `index.js` file so that when a change inside `print.js` is detected we tell webpack to accept the updated module.
 
 __index.js__
 
 ``` js
-import Library from './library';
+import _ from 'lodash';
+import printMe from './print.js';
 
 if (module.hot) {
-  module.hot.accept('./library', function() {
-    console.log('Accepting the updated library module!');
-    Library.log();
+    module.hot.accept('./print.js', function() {
+    console.log('Accepting the updated printMe module!');
+    printMe();
   })
 }
+
+function component() {
+  var element = document.createElement('div');
+  var btn = document.createElement('button');
+
+  element.innerHTML = _.join(['Hello', 'webpack'], ' ');
+
+  btn.innerHTML = 'Click me and check the console!';
+  btn.onclick = printMe;
+
+  element.appendChild(btn);
+
+  return element;
+}
+
+document.body.appendChild(component());
 ```
```

Comment:
You might be able to just leave out this whole section as nothing has changed from the `development` guide.